server/helpers/youtube-dl.ts

```diff
@@ -45,6 +51,35 @@ function getYoutubeDLInfo (url: string, opts?: string[]): Promise<YoutubeDLInfo>
   })
 }
 
+function getYoutubeDLSubs (url: string, opts?: object): Promise<YoutubeDLSubs> {
+  return new Promise<YoutubeDLSubs>((res, rej) => {
+    const cwd = CONFIG.STORAGE.TMP_DIR
+    const options = opts || { all: true, format: 'vtt', cwd }
+
+    safeGetYoutubeDL()
+      .then(youtubeDL => {
+        youtubeDL.getSubs(url, options, (err, files) => {
+          if (err) return rej(err)
+
+          const subtitles = files.reduce((acc, filename) => {
```

Comment:
Maybe a `.filter` based on the regex and a `.map` would be simpler?

Reply:
A `.reduce` seems more appropriate to match the language code ? 
With `.filter` and `.map` we'll need two `.filter` to validate extension / language code and to match the language code.

Reply:
oup's, you're right :+1: